src/components/progress/base.ts

```diff
@@ -54,36 +55,57 @@ export abstract class IgcProgressBaseComponent extends LitElement {
   @property({ attribute: 'label-format' })
   public labelFormat!: string;
 
-  @watch('indeterminate')
+  @watch('indeterminate', { waitUntilFirstUpdate: true })
   protected indeterminateChange() {
-    this.indeterminate ? this.animation?.cancel() : this.animation?.finish();
+    if (!this.indeterminate) {
+      this.runAnimation(0, this.value, true);
+    } else {
+      setTimeout(() => {
+        this.animation?.finish();
+        setTimeout(() => {
+          this.animation?.cancel();
+        }, 0);
+      }, 0);
```

Comment:
What exactly is the purpose of this thing?
Seems like a heavy workaround for something?

Reply:
If indeterminated is set to true while the value change animation is running, the two animations conflict with each other.

Reply:
I guess you could query the CSS animation and orchestrate the sync between the two when indeterminate is set